lib/metrics/reporter.ex

```diff
@@ -0,0 +1,64 @@
+defmodule Metrics.Reporter do
+  @moduledoc """
+  Behaviour for reporters forwarding values emitted by views to external systems
+
+  Currently there is no way to subscribe reporter only to specific metrics or views.
+  """
+
+  use GenServer
+
+  alias Metrics.{ReporterSupervisor, ReporterStore}
+
+  @doc """
+  Called whenever any of the views emits its values
+
+  The last argument is the UTC timestamp when the values have been emitted.
+  """
+  @callback handle_emit(Metrics.metric(), Metrics.view(), [Metrics.value()], NaiveDateTime.t()) ::
```

Comment:
I would keep the last argument the unix timestamp for simplicity. I even think Erlang's `logger` in OTP 21 did a similar change?